UPGRADE.md

```diff
@@ -3,6 +3,51 @@
 _`MAJOR` version bumps will have upgrade notes
 posted here._
 
+[2023-03-22] 7.x.x to 8.x.x
+---------------------------
+
+- Supported Python versions updated
```

Comment:
minor nit: do we want to use header (####), similar to the below upgrade guide, to make it neater?

Reply:
Made the headings bold!